src/core/wee-config.h

```diff
@@ -90,6 +90,17 @@ enum t_config_look_hotlist_sort
     CONFIG_LOOK_HOTLIST_SORT_NUMBER_DESC,
 };
 
+enum t_config_look_hotlist_get_sort
```

Comment:
Is it really necessary to have two separate enums for what's essentially the same thing?

Reply:
I didn't know I could reuse a configuration enum.
I'll look into how to do this.
Is it possible to make this option optional and when not set default to "same as look.hotlist.sort"?

Reply:
I can't seem to find an example of an optional option.
I dislike the idea of defaulting to any arbitrary value for `smart_jump`, because I believe many people assume it to follow the config of `hotlist_sort` by now.

Any idea how I would create an option for `smart_jump` that would contain the same options as `hotlist_sort` with an addition of `default` (that does the same as the selected `hotlist_sort`).

Reply:
It is possible to set a "parent" for an option: https://weechat.org/files/doc/devel/weechat_plugin_api.en.html#_config_new_option. That allows defaulting to another option's value when not set and seems like what you're thinking of. It's not totally automatic though: you have to check your option first and if it's unset use the other one yourself.

Internally the two options could be based on the same enum. To the user you need to manually write the possible choices strings in each option separately anyway, these can differ. If you set up the parent option, the "default" wouldn't be necessary and "null" would mean the same thing (use parent).

Reply:
Great, this sounds perfect.
Thank you!

Reply:
I'm not 100% sure about the documentation.

> name of option; with WeeChat ≥ 1.4, the name can include a parent option name (the value of parent option will be displayed in /set command output if this option is "null"), the syntax is then: "name << file.section.option"

Should the name of the option then be `"smart_jump_order << weechat.look.hotlist_sort"`?

Also as I said I could not find an example of an optional (nullable) option.
The macros I can find (`CONFIG_{INTEGER,STRING,BOOLEAN,COLOR}`) expect the value to be non-null. Should I create a new macro, say `CONFIG_IS_SET` or do something else?

Reply:
Yes, the option name should contain the parent, it's kind of a hack in WeeChat.

Each server's options have parents but that's kind of hard to understand as they're dynamic. Here's an example from a PR of mine: https://github.com/weechat/weechat/pull/684/files. The function you mentioned should be right. The macros are there for convenience, with parent/nullable options some functions need to be used directly.